### PR TITLE
Introduce ContentLanguages, remove ULS dependency

### PR DESCRIPTION
**Pull request by adrianheine**
https://phabricator.wikimedia.org/T86753

### CODE REVIEW

File: lib/util/util.ContentLanguages.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+		/**
+		 * Get a name for a given language code
+		 *
+		 * Preferebly, this should be in a language the user understands (i. e. the UI language).
+		 * Fallbacks are allowed, though. Returns null if the language code is not known.
+		 *
+		 * @return {string|null}
+		 */
+		getName: util.abstractMember
```
Comment (JeroenDeDauw) [inline lib/util/util.ContentLanguages.js]: Are these not implemented anywhere yet, or am I being blind?

File: lib/util/util.ContentLanguages.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	/**
+	 * A service storing informations about the languages available for the content
+	 * @class util.ContentLanguages
+	 * @abstract
+	 * @licence GNU GPL v2+
+	 * @author Adrian Heine <adrian.heine@wikimedia.de>
+	 *
+	 * @constructor
+	 */
+	util.ContentLanguages = function() {
```
Comment (JeroenDeDauw) [inline lib/util/util.ContentLanguages.js]: FYI: "util" is a code smell to me

File: lib/util/util.ContentLanguages.js (hunk 1 above)
Comment (adrianheine) [inline lib/util/util.ContentLanguages.js]: It's only implemented in Wikibase in https://gerrit.wikimedia.org/r/#/c/186229/.

File: lib/util/util.ContentLanguages.js (hunk 2 above)
Comment (adrianheine) [inline lib/util/util.ContentLanguages.js]: I agree. It's also not in any way the right place for a service interface.

File: lib/util/util.ContentLanguages.js (hunk 1 above)
Comment (JeroenDeDauw) [inline lib/util/util.ContentLanguages.js]: Oh right... that should have been kinda obvious since this commit is about breaking a dependency :)

Comment (adrianheine) [inline lib/util/util.ContentLanguages.js]: I could add a null implementation not limiting the languages used, though.

Comment (JeroenDeDauw) [inline lib/util/util.ContentLanguages.js]: If you think that is useful - I don't see a problem here. (Mind you, I did not do a full review either)

File: lib/jquery.ui/jquery.ui.languagesuggester.js [hunk 3]
```diff
@@ -1,15 +1,42 @@
 ( function( $ ) {
 	'use strict';
 
+var PARENT = $.ui.suggester;
```
Comment (thiemowmde) [inline lib/jquery.ui/jquery.ui.languagesuggester.js]: This could be it's own patch (along with your name change and maybe a few other lines in this patch), but is probably not worth it.

File: src/ExpertExtender/ExpertExtender.LanguageSelector.js [hunk 4]
```diff
@@ -7,28 +7,27 @@
...
 	 * @constructor
 	 *
+	 * @param {util.ContentLanguages} languages
 	 * @param {util.MessageProvider} messageProvider
 	 * @param {Function} getUpstreamValue
 	 * @param {Function} onValueChange
 	 */
-	ExpertExtender.LanguageSelector = function( messageProvider, getUpstreamValue, onValueChange ) {
+	ExpertExtender.LanguageSelector = function(
+		languages,
```
Comment (thiemowmde) [inline src/ExpertExtender/ExpertExtender.LanguageSelector.js]: Can you call this `contentLanguages` please?

File: src/ExpertExtender/ExpertExtender.LanguageSelector.js [hunk 5]
```diff
@@ -7,28 +7,27 @@
 	 * @class jQuery.valueview.ExpertExtender.LanguageSelector
 	 * @since 0.6
 	 * @licence GNU GPL v2+
-	 * @author Adrian Lang <adrian.lang@wikimedia.de>
+	 * @author Adrian Heine <adrian.heine@wikimedia.de>
 	 *
 	 * @constructor
 	 *
+	 * @param {util.ContentLanguages} languages
```
Comment (thiemowmde) [inline src/ExpertExtender/ExpertExtender.LanguageSelector.js]: I think this should be `{util.ContentLanguages|null}`.

**Comment by thiemowmde** [comment]
Did a code review, +1 from my side.

File: src/ExpertExtender/ExpertExtender.LanguageSelector.js (hunk 4 above)
Comment (adrianheine) [inline src/ExpertExtender/ExpertExtender.LanguageSelector.js]: Done.

File: src/ExpertExtender/ExpertExtender.LanguageSelector.js (hunk 5 above)
Comment (adrianheine) [inline src/ExpertExtender/ExpertExtender.LanguageSelector.js]: No, it's not allowed to be null.
